forum/qa-plugin/user-activity-log/user-activity-log-search.php

```diff
@@ -0,0 +1,177 @@
+        if(strpos($finalQuery, 'LIKE')){
+            $result = qa_db_query_sub($finalQuery, $this->searchArr['request'], $this->searchArr['date']."%", $this->searchArr['resultsCount']);
+        }else{
+            $result = qa_db_query_sub($finalQuery, $this->searchArr['request'], $this->searchArr['resultsCount']);
+        }
+       
+        if(mysqli_num_rows($result) <= $this->searchArr['resultsCount']){
+            $fullResultsArray = [];
+            while($row = mysqli_fetch_array($result, MYSQLI_ASSOC)){
+                $fullResultsArray[] = $row;
+            }
+            $this->dbResult = $fullResultsArray;
+            return 1;
+        }else{
+            $this->dbResult = $result;
+            return 0;
+        }
+
+    }
+
+    private function generateResultsTable()
+    {
+        $unfiltered = $this->dbResult;
+        if(qa_get_logged_in_level() < QA_USER_LEVEL_ADMIN){
+           $results = array_filter($unfiltered, function($field){
+                switch($field['event']){
+                    case 'q_vote_up': return ""; break;
+                    case 'q_vote_down': return ""; break;
+                    case 'q_vote_nil': return ""; break;
+                    case 'a_vote_up':return ""; break;
+                    case 'a_vote_down': return ""; break;
+                    case 'a_vote_nil': return ""; break;
+                    case 'c_vote_up': return ""; break;
+                    case 'c_vote_down': return ""; break;
+                    case 'c_vote_nil': return ""; break;
+                    case 'in_q_vote_up' :
+                    case 'in_a_vote_up' :
+                    case 'in_c_vote_up' :
+                    case 'q_flag':
+                    case 'a_flag':
+                    case 'c_flag':
+                    case 'q_unflag':
+                    case 'a_unflag':
+                    case 'c_unflag': return ""; break;
+                }
+                    return $field; 
+            });
+        }else{
+            $results = $unfiltered;
+        }
+
+        if(qa_get_logged_in_level() > QA_USER_LEVEL_EDITOR){
+            $tableHeader = 
+                "<table><tr><th>".qa_lang_html('user-activity-log/datetime').' </th>'.
+                    '<th> '.qa_lang_html('user-activity-log/ipaddress').'</th>'.
+                    '<th>'.qa_lang_html('user-activity-log/handle').'</th>'.
+                    '<th>'.qa_lang_html('user-activity-log/event').'</th>'.
+                "</tr>";
+
+            $tableContent = '';
+
+            foreach($results as $row){
+                $tableContent = $tableContent.'<tr>'.
+                    '<th>'.$row['datetime'].'</th>'.
+                    '<th> &nbsp;'.$row['ipaddress'].'</th>'.
+                    '<th>'.$row['handle'].'</th>'.
+                    '<th>'.qa_lang_html('user-activity-log/'.$row['event']).'</th>'.
+                    '</tr>';
+            }
+        }else if(qa_get_logged_in_level() == QA_USER_LEVEL_EDITOR){
```

Comment:
`===` zamiast `==`?

Reply:
Niezbyt.

`qa_get_logged_in_level()` zwraca typ string a ` QA_USER_LEVEL_EDITOR` to int, więc` ===` zwróci nam zawsze false

Reply:
Mhmm, to nie wiem na ile dobrą praktyką w PHP jest użycie `==`, a na ile lepiej samemu rzutować jedną z wartości do wspólnego typu.

Reply:
Z tego co znalazłem dobrą praktyką jest korzystanie z `===` przy sprawdzaniu czy wartość to null albo false ale nigdzie nie widziałem aby używanie `==` było złe